File: 8 laboratory/8b.py
input = open("input.txt")
n = int(input.readline())
table = [list(map(int, input.readline().split())) for i in range(n)]
for i in range(n):
    for j in range(n):
        if table[j][j] or table[j][i] != table[i][j]:
            open("output.txt", "w").write("NO")
            input = False
if input:
    open("output.txt", "w").write("YES")

File: 8 laboratory/8b.cpp
#include <fstream>
#include <vector>
using namespace std;

int main(){
    ifstream cin("input.txt");
    ofstream cout("output.txt");
    int n;
    cin >> n;
    vector <vector<int>> adjacency_matrix(n);
    for (auto &string: adjacency_matrix){
        string.resize(n);
        for (auto &element: string){
            cin >> element;
        }
    }
    for (int i = 0; i < n; i++) {
        for (int j = 0; j < n; j++) {
            if ((adjacency_matrix[j][j] == 1) or (adjacency_matrix[i][j] != adjacency_matrix[j][i])){
                cout << "NO";
                return 0;
            }
        }
    }
    cout << "YES";
    return 0;
}
